File: Python/beadsort/meta-llama-4-scout-17b-16e-instruct.py

import random

def beadsort(arr):
    """
    Beadsort algorithm implementation.

    :param arr: List of integers to be sorted
    :return: Sorted list of integers
    """
    # Find the maximum value in the array to determine the size of the beads array
    max_val = max(arr)

    # Initialize the beads array with zeros
    beads = [0] * (max_val + 1)

    # Count the occurrence of each value in the input array
    for val in arr:
        beads[val] += 1

    # Calculate cumulative counts
    for i in range(1, len(beads)):
        beads[i] += beads[i - 1]

    # Build the output array
    sorted_arr = [0] * len(arr)
    for val in arr:
        sorted_arr[beads[val] - 1] = val
        beads[val] -= 1

    return sorted_arr


import sys

def main():
    input_path = "inputs.txt" if len(sys.argv) < 2 else sys.argv[1]

    try:
        with open(input_path, 'r') as f:
            nums = []
            for line in f:
                line = line.strip()
                if line.isdigit():
                    nums.append(int(line))
    except FileNotFoundError:
        return

if __name__ == "__main__":
    main()

File: Java/beadsort/MetaLlama4Scout17b16eInstruct.java
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class MetaLlama4Scout17b16eInstruct {

    public static void beadsort(int[] arr) {
        if (arr.length == 0) return;

        int max = getMax(arr);
        int[] beads = new int[max + 1];

        for (int i : arr) {
            beads[i]++;
        }

        int j = 0;
        for (int i = 0; i < beads.length; i++) {
            while (beads[i] > 0) {
                arr[j] = i;
                j++;
                beads[i]--;
            }
        }
    }

    private static int getMax(int[] arr) {
        int max = arr[0];
        for (int i = 1; i < arr.length; i++) {
            if (arr[i] > max) max = arr[i];
        }
        return max;
    }

    public static void main(String[] args) {
        List<Integer> input = new ArrayList<>();

        try (java.io.BufferedReader br = new java.io.BufferedReader(new java.io.FileReader("../../inputs.txt"))) {
            String line;
            while ((line = br.readLine()) != null) {
                input.add(Integer.parseInt(line.trim()));
            }
        } catch (java.io.IOException e) {
            return;
        }

        // Convert List<Integer> to int[]
        int[] arr = input.stream().mapToInt(Integer::intValue).toArray();

        beadsort(arr);

        // Convert sorted int[] back to List<Integer> if needed
        List<Integer> sorted = new ArrayList<>();
        for (int num : arr) {
            sorted.add(num);
        }

    }
}

